src/modal/modal.scss

```diff
@@ -74,3 +75,40 @@
     transition-delay: 0s;
   }
 }
+
+@mixin iui-modal-full-page {
+  --iui-full-page-modal-transform-x: 100%;
+  --iui-full-page-modal-transform-y: 0%;
+
+  visibility: visible;
+  opacity: 1;
+  transform: translate(var(--iui-full-page-modal-transform-x), var(--iui-full-page-modal-transform-y));
+  @media (prefers-reduced-motion: no-preference) {
+    transition: transform $iui-speed ease-in;
+  }
+
+  > .iui-modal-dialog {
+    height: 100vh;
+    width: 100vw;
+    left: 0;
+    top: 0;
+    transform: none;
+    max-width: initial;
+    min-width: initial;
+    border-radius: 0;
+    box-shadow: none;
+
+    .iui-button-bar {
```

Comment:
If there is more content, button bar will go on top of it:
![image](https://user-images.githubusercontent.com/36186912/155507175-d6c5b7a6-260c-48d7-ae4a-686863878f1b.png)

Reply:
Fixed

Reply:
Ugh, I am not sure whether we can apply flex to the dialog because if someone has several inline elements next to each other, then we would break their layout:
![image](https://user-images.githubusercontent.com/36186912/155671478-a2afc163-da2d-4e8f-8457-af822c70af99.png)
I don't know why we didn't applied flex in the first place though... Maybe we could just apply flex only to full screen dialog.

Reply:
I like the idea of including `.iui-modal-content` in both default and full page modal.  All user content would need to go within this class.  I don't think the content inside there would have the issue you described, right?

Reply:
But users will need to do this change, otherwise it might break their current layouts.

Reply:
Yeah, I think it is the best approach right now.

Reply:
I've moved the flex and .iui-modal-content to the full-page-modal scss only.